app/scripts/controllers/edit/buildConfig.js

```diff
@@ -671,49 +683,16 @@ angular.module('openshiftConsole')
     };
 
     $scope.updateTriggers = function() {
-
-      // Helper for checking webhook objects in the buildConfig spec and creating if missing.
-      function reuseOrCreateWebhook(type) {
-        var webhooks = _.filter($scope.buildConfig.spec.triggers, function(obj) { return obj.type === type; });
-        if (_.isEmpty(webhooks)) {
-          var webhook = {
-            type: type
-          };
-          webhook[(type === "GitHub") ? "github" : "generic"] = {
-            secret: ApplicationGenerator._generateSecret()
-          };
-          webhooks.push(webhook);
-        }
-        return webhooks;
-      }
-
-      var presentTriggers = $scope.triggers.present;
-      var triggers = [];
-
-      if (presentTriggers.githubWebhook) {
-        triggers = triggers.concat(reuseOrCreateWebhook("GitHub"));
-      }
-
-      if (presentTriggers.genericWebhook) {
-        triggers = triggers.concat(reuseOrCreateWebhook("Generic"));
-      }
-
-      if (presentTriggers.configChange) {
-        triggers.push({
-          type: "ConfigChange"
-        });
-      }
-
-      // Add all imageChange triggers to the triggers array. The imageChange trigger for the builder is added in getTriggerMap()
-      // method into imageChangeTriggers array, even if the buildConfig doesn't contain it. If builder imageChange trigger is checked,
-      // keep the trigger, otherwise delete it.
-      triggers = triggers.concat($scope.triggers.imageChangeTriggers);
-      if (!presentTriggers.imageChange) {
-        _.remove(triggers, function(trigger) {
-          return trigger === $scope.triggers.builderImageChangeTrigger;
-        });
-      }
-
+      var triggers = [].concat($scope.triggers.githubWebhooks, 
+                              $scope.triggers.genericWebhooks, 
+                              $scope.triggers.imageChangeTriggers,
+                              $scope.triggers.builderImageChangeTrigger,
+                              $scope.triggers.configChangeTrigger);
+      // Filter webhook triggers that are not disabled or imageChange triggers that are present
+      triggers = _.filter(triggers, function(trigger) {
+        return trigger.disabled === false || trigger.present;
```

Comment:
We need to check if the `trigger.disabled` equals `false`, cause if the trigger wont have `disabled` field set  (eg. imageChange trigger) it will evaluate as `true`, which is not correct.